src/rustc.rs

```diff
@@ -19,20 +20,212 @@ impl TargetList {
 }
 
 pub trait VersionMetaExt {
-    fn host(&self) -> Host;
+    fn host(&self) -> TargetTriple;
     fn needs_interpreter(&self) -> bool;
 }
 
 impl VersionMetaExt for VersionMeta {
-    fn host(&self) -> Host {
-        Host::from(&*self.host)
+    fn host(&self) -> TargetTriple {
+        TargetTriple::from(&*self.host)
     }
 
     fn needs_interpreter(&self) -> bool {
         self.semver < Version::new(1, 19, 0)
     }
 }
 
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+pub struct QualifiedToolchain {
+    pub channel: String,
+    pub date: Option<String>,
+    pub(self) host: TargetTriple,
```

Comment:
`pub(self)` to ensure we don't touch it without using `QualifiedToolchain::replace_host` which ensures replacement is done properly